programs/drift/src/instructions/user.rs

```diff
@@ -1013,6 +1017,479 @@ pub fn handle_transfer_deposit<'c: 'info, 'info>(
+            user_authority: *authority_key,
+            user: from_user_key,
+            direction: DepositDirection::Deposit,
+            amount: borrow_amount,
+            oracle_price: borrow_from_oracle_price_data.price,
+            market_index: borrow_from_market_index,
+            market_deposit_balance: borrow_from_spot_market.deposit_balance,
+            market_withdraw_balance: borrow_from_spot_market.borrow_balance,
+            market_cumulative_deposit_interest: borrow_from_spot_market.cumulative_deposit_interest,
+            market_cumulative_borrow_interest: borrow_from_spot_market.cumulative_borrow_interest,
+            total_deposits_after: from_user.total_deposits,
+            total_withdraws_after: from_user.total_withdraws,
+            explanation: DepositExplanation::Transfer,
+            transfer_user: Some(to_user_key),
+        };
+        emit!(deposit_record);
+
+        to_user.increment_total_withdraws(
+            borrow_amount,
+            borrow_to_oracle_price_data.price,
+            borrow_to_spot_market.get_precision().cast()?,
+        )?;
+
+        controller::spot_position::update_spot_balances_and_cumulative_deposits_with_limits(
+            borrow_amount as u128,
+            &SpotBalanceType::Borrow,
+            &mut borrow_to_spot_market,
+            to_user,
+        )?;
+
+        let deposit_record_id = get_then_update_id!(borrow_to_spot_market, next_deposit_record_id);
+        let deposit_record = DepositRecord {
+            ts: clock.unix_timestamp,
+            deposit_record_id,
+            user_authority: *authority_key,
+            user: to_user_key,
+            direction: DepositDirection::Withdraw,
+            amount: borrow_amount,
+            oracle_price: borrow_to_oracle_price_data.price,
+            market_index: borrow_to_market_index,
+            market_deposit_balance: borrow_to_spot_market.deposit_balance,
+            market_withdraw_balance: borrow_to_spot_market.borrow_balance,
+            market_cumulative_deposit_interest: borrow_to_spot_market.cumulative_deposit_interest,
+            market_cumulative_borrow_interest: borrow_to_spot_market.cumulative_borrow_interest,
+            total_deposits_after: to_user.total_deposits,
+            total_withdraws_after: to_user.total_withdraws,
+            explanation: DepositExplanation::Transfer,
+            transfer_user: Some(from_user_key),
+        };
+        emit!(deposit_record);
+    }
+
+    drop(deposit_from_spot_market);
+    drop(deposit_to_spot_market);
+    drop(borrow_from_spot_market);
+    drop(borrow_to_spot_market);
+
+    from_user.meets_withdraw_margin_requirement_and_increment_fuel_bonus_swap(
+        &perp_market_map,
+        &spot_market_map,
+        &mut oracle_map,
+        MarginRequirementType::Initial,
+        deposit_from_market_index,
+        deposit_transfer.cast::<i128>()?,
+        borrow_from_market_index,
+        -borrow_amount.cast::<i128>()?,
```

Comment:
shouldn't the amount signs be swapped here? since you withdraw from `deposit_from` (negative), and deposit to `borrow_from` (positive)